pkg/debug_backup.go

```diff
@@ -0,0 +1,314 @@
+)
+
+func NewCmdDebugBackup() *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:               "backup",
+		Short:             `Debug backup`,
+		Long:              `Debug backup by describing and showing logs of backup resources`,
+		Example:           debugBackupExample,
+		DisableAutoGenTag: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if backupConfig == "" && backupBatch == "" {
+				return fmt.Errorf("neither BackupConfiguration nor BackupBatch name has been provided")
+			}
+
+			if err := showVersionInformation(); err != nil {
+				return err
+			}
+
+			if backupConfig != "" {
+				bc, err := stashClient.StashV1beta1().BackupConfigurations(namespace).Get(context.TODO(), backupConfig, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				if err := debugBackupConfig(bc); err != nil {
+					return err
+				}
+			} else {
+				bb, err := stashClient.StashV1beta1().BackupBatches(namespace).Get(context.TODO(), backupBatch, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				if err := debugBackupBatch(bb); err != nil {
+					return err
+				}
+
+			}
+			return nil
+		},
+	}
+	cmd.Flags().StringVar(&backupConfig, "backupconfig", backupConfig, "Name of the BackupConfiguration to debug")
+	cmd.Flags().StringVar(&backupBatch, "backupbatch", backupBatch, "Name of the BackupBatch to debug")
+	return cmd
+}
+
+func showVersionInformation() error {
+	klog.Infoln("\n\n\n===============[ Client and Server version information ]===============")
+	if err := sh.Command("kubectl", "version", "--short").Run(); err != nil {
+		return err
+	}
+
+	pod, err := GetOperatorPod()
+	if err != nil {
+		return err
+	}
+	klog.Infoln("\n\n\n===============[ Operator's binary version information ]===============")
+
+	if err := sh.Command("kubectl", "exec", "-it", "-n", pod.Namespace, pod.Name, "-c", "operator", "--", "/stash-enterprise", "version").Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func debugBackupConfig(backupConfig *v1beta1.BackupConfiguration) error {
+	if backupConfig.Status.Phase == v1beta1.BackupInvokerReady {
+		backupSessions, err := getOwnedBackupSessions(backupConfig)
+		if err != nil {
+			return err
+		}
+		jobList, podList, err := getAllJobsAndPods()
+		if err != nil {
+			return err
+		}
+		for _, backupSession := range backupSessions {
+			if backupSession.Status.Phase == v1beta1.BackupSessionFailed {
+				if err := describeObject(&backupSession, v1beta1.ResourceKindBackupSession); err != nil {
+					return err
+				}
+				backupModel := util.BackupModel(backupConfig.Spec.Target.Ref.Kind)
+				if backupModel == apis.ModelSidecar {
+					if err := debugWorkloadPods(backupConfig.Spec.Target.Ref, apis.StashContainer); err != nil {
+						return err
+					}
+
+				} else if backupModel == apis.ModelCronJob {
+					if err := debugBackupJob(&backupSession, jobList, podList); err != nil {
+						return err
+					}
+				}
+			}
+		}
+	} else {
```

Comment:
Refactor this function. `debugBackupConfig` should be responsible only for describing `BackupConfiguration`.